lib/services/cups.pm

```diff
@@ -18,31 +18,47 @@ use utils;
 use strict;
 use warnings;
 
+my $service_type = 'Systemd';
+
 sub install_service {
     zypper_call("in cups", exitcode => [0, 102, 103]);
 }
 
 sub config_service {
-    script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
-    validate_script_output 'cupsd -t', sub { m/is OK/ };
+    if ($service_type eq 'Systemd') {
+        script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
+        validate_script_output 'cupsd -t', sub { m/is OK/ };
+    } else {
+        # Allow new printers to be added using device URIs "file:/filename"
+        script_run 'echo FileDevice Yes >> /etc/cups/cupsd.conf';
+        assert_script_run 'cupsd';
+    }
 }
 
 sub enable_service {
-    systemctl 'enable cups.service';
+    common_service_action 'cups', $service_type, 'enable';
 }
 
 sub start_service {
-    systemctl 'start cups.service';
+    common_service_action 'cups', $service_type, 'start';
 }
 
 # check service is running and enabled
 sub check_service {
-    systemctl 'is-enabled cups.service';
-    systemctl 'is-active cups';
+    common_service_action 'cups', $service_type, 'is-enabled';
+    common_service_action 'cups', $service_type, 'is-active';
 }
 
 # check cups function
 sub check_function {
+    # if we migrated from sle11sp4, we need change the configure file before check
+    # cups function.
+    if ((get_var('ORIGIN_SYSTEM_VERSION') eq '11-SP4') && ($service_type eq 'Systemd')) {
```

Comment:
The logic seems conflict, 11sp4 support systemd ?  
I remember you already set service_type in lib/service_check.pm, but now you start check ORIGIN_SYSTEM_VERSION again?

Reply:
This is for migration from 11sp4 to  15sp3.  use this to differ the others that migration from systemd to sles15sp3

Reply:
service_type --> hdd_base_version  --> HDDVERSION   if HDDVERSION will change, then we need update service_type logic
check system whether migration from 11sp4 or not, we either using ORIGIN_SYSTEM_VERSION or service_type, do not mix them

Reply:
now we use the $hdd_base_version from the service_check.pm to show which base version was we migration from.

Reply:
it's works and updated the test results here. could you please take a  look again.